client.go

```
@@ -126,23 +121,27 @@ func (c *Client) authenticate() (*accessToken, error) {
 	return nil, fmt.Errorf("Unable to find token for '%s'\nTokens returned: %+v\n", c.Company, tokens)
 }
 
-// func (c *Client) CreateAttack(ac AttackCommand, token accessToken) error {
-// rurl := c.resourceURL("attacks/new")
-//
-// jsonBody, err := json.Marshal(&ac)
-// if err != nil {
-// 	return fmt.Errorf("Failed to marshal attack command: %v", err)
-// }
-
-// create new request
-// add headers
-// ignore response body for now
-
-// return nil
-// }
-
 // resourceURL safely joins a string path (e.g. "my/resource") to an existing URL.
 func (c *Client) resourceURL(path string) *url.URL {
 	rel := &url.URL{Path: path}
 	return c.BaseURL.ResolveReference(rel)
 }
+
+// dispatchRequest to server and return a byte slice containing the response body.
+// An error will be returned instead if the request fails for some reason.
+func (c *Client) dispatchRequest(req *http.Request) ([]byte, error) {
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Request failed: %v", err)
+	}
+
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
```

Don't think we should ignore this error.